### ISSUE TITLE
Google Assistant Integration

### ISSUE THREAD

**Issue by pedrofracassi** [SwitchbladeBot/switchblade#577]
Now that we can have webhooks on the bot (thanks, dokku!), we could try to make some integrations to Google Assistant:

`Ok google, ask Switchblade to play darude sandstorm`

`Ok google, set the volume to 150`

We would obviously have to somehow authenticate the users, just like DBL does.

**Comment by justandras** [comment]
Nah, Just play it on all servers when someone asks for it. or maybe on a random server. That would be fun 😃

**Comment by pedrofracassi** [comment]
[sure, why not?](https://cdn.frankerfacez.com/emoticon/175963/4)

**Comment by dpaiv0** [comment]
see [Google Assistant Developer Community Program](https://developers.google.com/actions/community/overview)

**Comment by pedrofracassi** [comment]
That's what I'm aiming for

**Comment by metehus** [comment]
Could use the connections from #748 to connect the user to his google account

**Comment by pedrofracassi** [comment]
Yup. We could also let users login to discord right from Google Assistant if they prefer to. DBL does that.

**Comment by metehus** [comment]
yes and this will make a connection

**Comment by metehus** [comment]
> yes and this will make a connection

wait, in the g assistant it will only save the id i think, so it wont be a connection or anything

**Comment by dpaiv0** [comment]
No? You can store whatever you want

**Comment by dpaiv0** [comment]
https://developers.google.com/actions/assistant/save-data

**Comment by metehus** [comment]
> No? You can store whatever you want

@dpaiv0 no, i mean like, it will be saving the discord user id on the g assistant, and not switch will be saving the google account connected

**Comment by pedrofracassi** [comment]
Ok, serious talk now. I've finally got a Google Home Mini, and am now motivated to get this to work.

**Comment by pedrofracassi** [comment]
We'll need to create our own oAuth thing, and use the tokens generated by it to authenticate on a few new webhooks we'll create:

- `/webhooks/music/play`
- `/webhooks/music/pause` (also used for resume, but this might be a bad idea)
- `/webhooks/music/shuffle`
- `/webhooks/music/skip`
- `/webhooks/music/bassboost`

These will be used by Tensorflow to tell the bot when to do stuff. We'll then use the oAuth token sent by TF to figure out which user is performing the action, and in which voice channel they're currently in.

**Comment by pedrofracassi** [comment]
https://dialogflow.com/docs/getting-started
